plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1122,6 +1113,28 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
 
       my->chain.emplace( *my->chain_config, std::move(pfs), *chain_id );
 
+      if( options.count( "transaction-retry-max-storage-size-gb" )) {
+         EOS_ASSERT( options.at( "disable-api-persisted-trx" ).as<bool>(), plugin_config_exception,
+                     "disable-api-persisted-trx must be set to true for transaction retry feature" );
+         EOS_ASSERT( !options.count( "producer-name"), plugin_config_exception,
```

Comment:
Is there a technical reason this cannot be run on a producer node? @heifner 

I have a use case where we run a blockchain locally for unit testing purposes and it would be too much to expect the user to spin up another API node just for this purpose.

Reply:
The transaction retry feature re-sends a transaction to the network when it does not observe a successful locally speculatively executed transaction making it into a produced block. If you are running a single block production test network then the feature is not needed on the block producer because a transaction will either fail or succeed and be placed into a block.